src/super_gradients/training/models/detection_models/csp_resnet.py

```diff
@@ -0,0 +1,317 @@
+        y = self.conv1(x)
+        y = self.conv2(y)
+        if self.shortcut:
+            return x + y
+        else:
+            return y
+
+
+class EffectiveSELayer(nn.Module):
+    """Effective Squeeze-Excitation
+    From `CenterMask : Real-Time Anchor-Free Instance Segmentation` - https://arxiv.org/abs/1911.06667
+    """
+
+    def __init__(self, channels: int):
+        super(EffectiveSELayer, self).__init__()
+        self.fc = nn.Conv2d(channels, channels, kernel_size=1, padding=0)
+        self.act = nn.Hardsigmoid(inplace=True)
+
+    def forward(self, x):
+        x_se = x.mean((2, 3), keepdim=True)
+        x_se = self.fc(x_se)
+        return x * self.act(x_se)
+
+
+class CSPResStage(nn.Module):
+    def __init__(
+        self,
+        block_fn: Type[nn.Module],
+        in_channels: int,
+        out_channels: int,
+        num_blocks,
+        stride: int,
+        activation_type: Type[nn.Module],
+        use_attention: bool = True,
+        use_alpha: bool = False,
+    ):
+        super().__init__()
+
+        mid_channels = (in_channels + out_channels) // 2
+        if stride == 2:
+            self.conv_down = ConvBNLayer(
+                in_channels, mid_channels, 3, stride=2, padding=1, activation_type=activation_type
+            )
+        else:
+            self.conv_down = None
+        self.conv1 = ConvBNLayer(
+            mid_channels, mid_channels // 2, kernel_size=1, stride=1, padding=0, activation_type=activation_type
+        )
+        self.conv2 = ConvBNLayer(
+            mid_channels, mid_channels // 2, kernel_size=1, stride=1, padding=0, activation_type=activation_type
+        )
+        self.blocks = nn.Sequential(
+            *[
+                block_fn(
+                    mid_channels // 2,
+                    mid_channels // 2,
+                    activation_type=activation_type,
+                    shortcut=True,
+                    use_alpha=use_alpha,
+                )
+                for _ in range(num_blocks)
+            ]
+        )
+        if use_attention:
+            self.attn = EffectiveSELayer(mid_channels)
+        else:
+            self.attn = nn.Identity()
+
+        self.conv3 = ConvBNLayer(
+            mid_channels, out_channels, kernel_size=1, stride=1, padding=0, activation_type=activation_type
+        )
+
+    def forward(self, x):
+        if self.conv_down is not None:
+            x = self.conv_down(x)
+        y1 = self.conv1(x)
+        y2 = self.blocks(self.conv2(x))
+        y = torch.concat([y1, y2], dim=1)
+        y = self.attn(y)
+        y = self.conv3(y)
+        return y
+
+
+class CSPResNet(nn.Module):
+    def __init__(
+        self,
+        layers=[3, 6, 6, 3],
+        channels=[64, 128, 256, 512, 1024],
+        activation_type: Type[nn.Module] = torch.nn.SiLU,
+        return_idx=[1, 2, 3],
+        use_large_stem: bool = True,
+        width_mult=1.0,
+        depth_mult=1.0,
+        use_alpha: bool = False,
+    ):
```

Comment:
please add documentation

Reply:
please extract the default values into a yaml to at least a dictionary (to be later moved to yaml)

Reply:
you can use your ActivationsTypeFactory with `resolve_param('activation_type', ActivationsTypeFactory)`

Reply:
> please extract the default values into a yaml to at least a dictionary (to be later moved to yaml)
Can you please elaborate on this?
CSPResNet was mean to be instantiated inside PPYolo, this is where extraction of params from config dict may take place. Would that work?

Reply:
regarding the parameters extraction, i meant that prefer not to have default parameters inside the code. many of our models already have those, but we plan on moving them all to yaml. We did the same process on DataLoaders a few weeks ago, so you can see now that dataloaders params are not defined as default values in the init().
the idea is to expose which parameters each block except, all in one place (the yaml). 

regarding the `resolve_param()`, I think it would be best to add it both here and in PP-Yolo. If an nn.Module is passed from PP-Yolo, it will just pass through, but if someone wants to use CSPResNet as the backbone to anything else (or even on its own), it will translate a string into nn.Module

Reply:
TY, I've added corresponding YAML files and indeed code became much cleaner :)